target_chains/sui/contracts/sources/pyth.move

```diff
@@ -478,108 +599,112 @@ module pyth::pyth_tests{
+                        price::new(i64::new(1508, false), 3, i64::new(5, true), 1663680745),
+                    ),
+                ),
+            ]
+    }
+
+    #[test_only]
+    /// Compare the expected price feed with the actual Pyth price feeds.
+    fun check_price_feeds_cached(expected: &vector<PriceInfo>, actual: &vector<PriceInfoObject>) {
+        // Check that we can retrieve the correct current price and ema price for each price feed
+        let i = 0;
+        while (i < vector::length(expected)) {
+            let price_feed = price_info::get_price_feed(vector::borrow(expected, i));
+            let price = price_feed::get_price(price_feed);
+            let ema_price = price_feed::get_ema_price(price_feed);
+            let price_identifier = price_info::get_price_identifier(vector::borrow(expected, i));
+
+            let actual_price_info = price_info::get_price_info_from_price_info_object(vector::borrow(actual, i));
+            let actual_price_feed = price_info::get_price_feed(&actual_price_info);
+            let actual_price = price_feed::get_price(actual_price_feed);
+            let actual_ema_price = price_feed::get_ema_price(actual_price_feed);
+            let actual_price_identifier = price_info::get_price_identifier(&actual_price_info);
+
+            assert!(price == actual_price, 0);
+            assert!(ema_price == actual_ema_price, 0);
+            assert!(price_identifier::get_bytes(&price_identifier) == price_identifier::get_bytes(&actual_price_identifier), 0);
+
+            i = i + 1;
+        };
+    }
+
+    #[test]
+    fun test_get_update_fee() {
+        let single_update_fee = 50;
+        let initial_guardians = vector[x"beFA429d57cD18b7F8A4d91A2da9AB4AF05d0FBe"];
+        let (scenario, test_coins, _clock) =  setup_test(500, 23, x"5d1f252d5de865279b00c84bce362774c2804294ed53299bc4a0389a5defef92", vector[], initial_guardians, single_update_fee, 0);
+        test_scenario::next_tx(&mut scenario, DEPLOYER, );
+        let pyth_state = take_shared<PythState>(&scenario);
+        // Pass in a single VAA
+
+        let single_vaa = vector[
+            x"fb1543888001083cf2e6ef3afdcf827e89b11efd87c563638df6e1995ada9f93",
+        ];
+
+        assert!(pyth::get_total_update_fee(&pyth_state, vector::length<vector<u8>>(&single_vaa)) == single_update_fee, 1);
+
+        let multiple_vaas = vector[
+            x"4ee17a1a4524118de513fddcf82b77454e51be5d6fc9e29fc72dd6c204c0e4fa",
+            x"c72fdf81cfc939d4286c93fbaaae2eec7bae28a5926fa68646b43a279846ccc1",
+            x"d9a8123a793529c31200339820a3210059ecace6c044f81ecad62936e47ca049",
+            x"84e4f21b3e65cef47fda25d15b4eddda1edf720a1d062ccbf441d6396465fbe6",
+            x"9e73f9041476a93701a0b9c7501422cc2aa55d16100bec628cf53e0281b6f72f"
+        ];
+
+        // Pass in multiple VAAs
+        assert!(pyth::get_total_update_fee(&pyth_state, vector::length<vector<u8>>(&multiple_vaas)) == 250, 1);
```

Comment:
nitpick: `5 * single_update_fee` would be an easier test condition for the reader to understand